scripts/SubsetSelection/src/encoders/openai.py

```
@@ -0,0 +1,92 @@
+        self.model = self.initialize_model(model_name, device)
+        self.max_attempts = max_attempts
+        self.timeout = timeout
+
+    def get_api_key(self):
+        load_dotenv()
+        api_key = os.getenv("OPENAI_API_KEY")
+        if not api_key:
+            raise ValueError("OPENAI_API_KEY not found in environment variables")
+        return api_key
+
+    def initialize_model(self, model_name, device, use_fp16=False):
```

Ruff (0.13.1)

Unused method argument: `device`

(ARG002)